examples/03_connectivity/plot_rest_clustering.py

```diff
@@ -30,130 +52,144 @@
 
-# Unmask the labels
+# Now, ward_labels_img are Nifti1Image object, it can be saved to file
+# with the following code:
+ward_labels_img.to_filename('ward_parcellation.nii.gz')
 
-# Avoid 0 label
-labels = ward.labels_ + 1
-labels_img = nifti_masker.inverse_transform(labels)
+from nilearn import plotting
+from nilearn.image import mean_img, index_img
 
-from nilearn.image import mean_img
-mean_func_img = mean_img(func_filename)
+first_plot = plotting.plot_roi(ward_labels_img, title="Ward parcellation",
+                               display_mode='xz')
 
+# Grab cut coordinates from this plot to use as a common for all plots
+cut_coords = first_plot.cut_coords
+###########################################################################
+# Compressed representation of Ward clustering
+# --------------------------------------------
+#
+# Second, we illustrate the effect that the clustering has on the signal.
+# We show the original data, and the approximation provided by the
+# clustering by averaging the signal on each parcel.
 
-first_plot = plot_roi(labels_img, mean_func_img, title="Ward parcellation",
-                      display_mode='xz')
+# Grab number of voxels from attribute mask image (mask_img_).
+import numpy as np
+original_voxels = np.sum(ward.mask_img_.get_data())
 
-# common cut coordinates for all plots
-cut_coords = first_plot.cut_coords
+# Compute mean over time on the functional image to use the mean
+# image for compressed representation comparisons
+mean_func_img = mean_img(dataset.func[0])
 
-##################################################################
-# labels_img is a Nifti1Image object, it can be saved to file with the
-# following code:
-labels_img.to_filename('parcellation.nii')
+# Compute common vmin and vmax
+vmin = np.min(mean_func_img.get_data())
+vmax = np.max(mean_func_img.get_data())
 
+plotting.plot_epi(mean_func_img, cut_coords=cut_coords,
+                  title='Original (%i voxels)' % original_voxels,
+                  vmax=vmax, vmin=vmin, display_mode='xz')
 
-##################################################################
-# Second, we illustrate the effect that the clustering has on the
-# signal. We show the original data, and the approximation provided by
-# the clustering by averaging the signal on each parcel.
-#
-# As you can see below, this approximation is very good, although there
+# A reduced data can be create by taking the parcel-level average:
+# Note that, the Parcellations object with any method has opportunity to
+# use a transform method that modifies input features. Here it reduces their
+# dimension. Note that we fit before calling a transform so that average signals
+# can be created on the brain parcellations with fit call.
+fmri_reduced = ward.transform(dataset.func)
+
+# Display the corresponding data compressed using the parcellation using
+# parcels=2000.
+fmri_compressed = ward.inverse_transform(fmri_reduced)
+
+plotting.plot_epi(index_img(fmri_compressed[0], 0),
```

Comment:
I don't understand why we need both "index_img" and the list indexing on "fmi_compressed". It seems that fmri_compressed is not what I would have expected it to be: a 4D nifti image.

Reply:
This plot is to show the compressed version in comparison to a 4D nifti image which is on above.

I should either use index_img or mean_img for 3D visualization with ```plot_epi```.

Reply:
>the list indexing on "fmi_compressed"

Because this object operates on a list of nifti images and in return as a list.